test/ffmpeg-vaapi/vpp/contrast.py

```diff
@@ -8,33 +8,50 @@
 from ..util import *
 
 spec = load_test_spec("vpp", "contrast")
+spec_r2r = load_test_spec("vpp", "contrast", "r2r")
 
 _NOOP_LEVEL = 10 # i.e. 1.0 in ffmpeg range
 
-@slash.requires(have_ffmpeg)
-@slash.requires(have_ffmpeg_vaapi_accel)
-@slash.requires(*have_ffmpeg_filter("procamp_vaapi"))
-@slash.parametrize(*gen_vpp_contrast_parameters(spec))
-@platform_tags(VPP_PLATFORMS)
-def test_default(case, level):
-  params = spec[case].copy()
-  params.update(
+def init(tspec, case, level):
+  tparams = tspec[case].copy()
+  tparams.update(
     level = level, mlevel = mapRange(level, [0, 100], [0.0, 10.0]),
-    mformat = mapformat(params["format"]))
-  params["ofile"] = get_media()._test_artifact(
-    "{}_contrast_{level}_{format}_{width}x{height}"
-    ".yuv".format(case, **params))
+    mformat = mapformat(tparams["format"]))
+
+  if tparams["mformat"] is None:
+    slash.skip_test("{format} format not supported".format(**tparams))
 
-  if params["mformat"] is None:
-    slash.skip_test("{format} format not supported".format(**params))
+  return tparams
 
+def call_ffmpeg(params):
   call(
     "ffmpeg -hwaccel vaapi -vaapi_device /dev/dri/renderD128 -v debug"
     " -f rawvideo -pix_fmt {mformat} -s:v {width}x{height} -i {source}"
     " -vf 'format=nv12,hwupload,procamp_vaapi=c={mlevel}"
     ",hwdownload,format=nv12'"
     " -pix_fmt {mformat} -vframes {frames} -y {ofile}".format(**params))
 
+def gen_output(case, params):
+  params["ofile"] = get_media()._test_artifact(
+    "{case}_contrast_{level}_{format}_{width}x{height}".format(case = case, **params))
+
+  if params.get("r2r", None) is not None:
+     params["ofile"] =  params["ofile"] + "_r2r"
+
+  params["ofile"] = params["ofile"] + ".yuv"
```

Comment:
Do not change the filename that you registered with `get_media()._test_artifact()`.  Build the filename first, then register it with  `get_media()._test_artifact()` last.

Reply:
For `get_media()._test_artifact()`:  Basically, you're telling the media plugin that the test is going to produce an artifact file named, say, `abc`.  When the test finishes, the media plugin will automatically remove/save `abc` depending on the artifact-retention setting.  But, if the test actually creates an artifact called `abc_xyz.yuv`, then you've bypassed the media plugin artifact retention mechanism altogether and the file stays around.

Reply:
ok thanks, got the context of it, will correct it

Reply:
corrected the filename with get test artifact